### PR TITLE
Main documentation: simplify links and reflow text

### PR DESCRIPTION
**Pull request by pitdicker**
It seems out main documentation needs a good rewrite, not everything matches what we would recommend today.
But for now I just simplified the links and reflowed the surrounding text.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,21 +1,23 @@
 //! # Chrono: Date and Time for Rust
 //!
 
-//! Chrono aims to provide all functionality needed to do correct operations on dates and times in the
-//! [proleptic Gregorian calendar](https://en.wikipedia.org/wiki/Proleptic_Gregorian_calendar):
+//! Chrono aims to provide all functionality needed to do correct operations on dates and times in
```
Comment (djc) [inline src/lib.rs:4]: How did you reflow? It doesn't look very consistent.

Comment (pitdicker) [reply src/lib.rs:3]: Manually against a line at the 100-characters column in my editor. And sorry, it should be a bit better now. I missed a few lines, and later changes to a few links messed up parts that I did before.

File: src/lib.rs [hunk 2]
```diff
@@ -35,16 +36,19 @@
 //!
 //! - `serde`: Enable serialization/deserialization via [serde].
 //! - `rkyv`: Deprecated, use the `rkyv-*` features.
-//! - `rkyv-16`: Enable serialization/deserialization via [rkyv], using 16-bit integers for integral `*size` types.
-//! - `rkyv-32`: Enable serialization/deserialization via [rkyv], using 32-bit integers for integral `*size` types.
-//! - `rkyv-64`: Enable serialization/deserialization via [rkyv], using 64-bit integers for integral `*size` types.
+//! - `rkyv-16`: Enable serialization/deserialization via [rkyv],
```
Comment (djc) [inline src/lib.rs:39]: In particular, why use the shorter lines here?

Comment (pitdicker) [reply src/lib.rs:39]: The alternative would be
```rust
//! - `rkyv-16`: Enable serialization/deserialization via [rkyv], using 16-bit integers for integral
//!   `*size` types.
```

A manual wrap in this case seemed a little nicer.

Comment (djc) [reply src/lib.rs:39]: Fair enough.

**Review by djc — APPROVED** [review]

**Pull request merged**